### PR TITLE
Add support for NUnit4

### LINKED ISSUE #2482: Add support for NUnit4
### Background and motivation

[NUnit 4.0.0](https://docs.nunit.org/articles/nunit/release-notes/framework.html#nunit-400---november-26-2023) got released on Nov 26th.
As we already support NUnit3 we should also support the latest version.

### Alternative Concerns

_No response_

### Are you willing help with a pull-request?

Yes, please assign this issue to me.

### PR DESCRIPTION
**Pull request by vbreuss**
Add FrameworkSpecs for [NUnit4](https://docs.nunit.org/articles/nunit/release-notes/framework.html#nunit-400---november-26-2023) and mention support in documentation and package information.

*Fixes #2482*

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [x] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by dennisdoomen** [comment]
Since it seems to contain some breaking changes, supporting both Nunit 3 and 4 makes sense to me.

File: Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace NUnit4.Specs;
+
+[TestFixture]
+public class FrameworkSpecs
+{
+    [Test]
+    public void When_nunit4_is_used_it_should_throw_nunit_exceptions_for_assertion_failures()
```
Comment (dennisdoomen) [inline Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs:11]: 🔧 Please use the new namign convetion.

File: Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        // Act
+        Action act = () => 0.Should().Be(1);
+
+        // Assert
+        Exception exception = act.Should().Throw<Exception>().Which;
+        exception.GetType()
+            .FullName.Should()
+            .ContainEquivalentOf("NUnit.Framework.AssertionException");
```
Comment (dennisdoomen) [inline Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs:20]: 🤔 Are you sure this is the right one? I was also expecting a different version of the `NUnitTestFramework` type. Or are the exception types the same?

Comment (jnyrup) [reply Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs:20]: Our current `NUnitTestFramework` also matches NUnit4, so no changes are needed on our side.

[NUnit.Framework.AssertionException](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Exceptions/AssertionException.cs)

image

**Comment by jnyrup** [comment]
NUnit4 needs to added here
https://github.com/fluentassertions/fluentassertions/blob/29e969944d74182432e620652d8a27d96d6bf77d/Build/Build.cs#L275-L281

File: Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs (hunk 2 above)
Comment (vbreuss) [reply Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs:20]: The Test adapter also remained unchanged, according to the [release notes](https://docs.nunit.org/articles/nunit/release-notes/framework.html#nunit-400---november-26-2023):
image

File: Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs (hunk 1 above)
Comment (vbreuss) [reply Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs]: Implemented in [0c4712f](https://github.com/fluentassertions/fluentassertions/pull/2483/commits/0c4712fa52b1cecaa9b6c74475bbcd33b8ff1faa).

**Comment by jnyrup** [comment]
The failing build is because NUnit4.Specs needs to be build in debug for the `CI` solution configuration.

**Review by dennisdoomen — APPROVED** [review]

**Review by jnyrup — APPROVED** [review]

**Pull request merged**